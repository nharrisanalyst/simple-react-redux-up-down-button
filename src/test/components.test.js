import {shallow} from 'enzyme';
import UpDownButtons from '../components/upDownButtons'
import React from 'react'
import renderer from 'react-test-renderer';
import sinon from 'sinon';

describe('<UpDownButtons/>',()=>{
  const wrapper = shallow(<UpDownButtons />);
  
  it('renders correctly', () =>{
   const tree = renderer.create(
    <UpDownButtons current='0' />
  ).toJSON();
  expect(tree).toMatchSnapshot();
  
  })

  
  it('renders two buttons', () =>{
  
  

  expect(wrapper.find('button')).toHaveLength(2)
  
  });
  
  it('when button is clicked it is called #buttonUp', ()=>{
   const mockHandle=sinon.spy();
   const wrapper = shallow(<UpDownButtons handleClickUp={mockHandle} />);
   
   
   const button = wrapper.find('#buttonUp')
   
   button.simulate('click')
  expect(mockHandle.calledOnce).toEqual(true)
  
  });

  it('when button is clicked it is called #buttonDown', ()=>{
   const mockHandle=sinon.spy();
   const wrapper = shallow(<UpDownButtons handleClickDown={mockHandle} />);
   
   
   const button = wrapper.find('#buttonDown')
   
   button.simulate('click')
  expect(mockHandle.calledOnce).toEqual(true)
  
  });
  
  it('when button is  clicked #buttonUp moves the text up one',()=>{
    const mockHandle=sinon.spy();
    const wrapper = shallow(<UpDownButtons/>)
    const current =wrapper.prop('current')
    console.log(wrapper)
         
  
  });

})