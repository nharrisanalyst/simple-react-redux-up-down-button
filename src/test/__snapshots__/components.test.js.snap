// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpDownButtons/> renders correctly 1`] = `
<div
  style={
    Object {
      "margin": "10px",
      "textAlign": "center",
    }
  }
>
  <button
    className="button"
    id="buttonUp"
    onClick={[Function]}
    style={
      Object {
        "height": "25px",
        "margin": "5px",
        "width": "35px",
      }
    }
  >
    +1
  </button>
  0
  <button
    className="button"
    id="buttonDown"
    onClick={[Function]}
    style={
      Object {
        "height": "25px",
        "margin": "5px",
        "width": "35px",
      }
    }
  >
    -1
  </button>
</div>
`;
